cloudify_agent/installer/script.py

```diff
@@ -203,3 +207,27 @@ def install_script_path(cloudify_agent):
         yield script_path
     finally:
         os.remove(script_path)
+
+
+def _cleanup_after_installation(path):
+    """Mark path to be deleted after agent installation.
+
+    This simply adds the path to cloudify_agent inside runtime properties,
+    so that it can be removed later.
+    """
+    cloudify_agent = ctx.instance.runtime_properties.get('cloudify_agent', {})
+    cleanup = cloudify_agent.get(LOCAL_CLEANUP_PATHS_KEY, [])
+    cleanup.append(path)
+    cloudify_agent[LOCAL_CLEANUP_PATHS_KEY] = cleanup
+    ctx.instance.runtime_properties['cloudify_agent'] = cloudify_agent
+
+
+def cleanup_scripts():
+    """Remove the files that were scheduled for deletion."""
+    cloudify_agent = ctx.instance.runtime_properties.get('cloudify_agent', {})
+    paths = cloudify_agent.pop(LOCAL_CLEANUP_PATHS_KEY, [])
```

Comment:
Does this pop gets registered by the ctx and the runtime props?

Reply:
it doesn't, added a `update` there